### DISCUSSION TITLE
Ready for the release of 1.2.1 ?

### DISCUSSION THREAD

**Discussion by bcoconni** [JSBSim-Team/jsbsim#1133] [General]
I am about to push the button of the release `v1.2.1`. I have compiled the changelog below.
Is there anything important that is missing or should be included in this release ?

### User
* Fixed a bug in the computation of orbital parameters where the perigee argument was valid only if the inclination and eccentricity were non-zero (PR #981)
* Restored the ability to use the linearization feature from the console and from XML scripts (PR #991). As a result of this change, the script [737_cruise_steady_turn_simplex.xml](https://github.com/JSBSim-Team/jsbsim/blob/e5192df5b7e6f4003fdc351bde17d38dfc6b1bff/scripts/737_cruise_steady_turn_simplex.xml) no longer fails.
* Fixed an error in the computation of rotation rates in the body frame (aka `vPQRdot`) (regression test added by PR #1034 and fixed by PR #1036)
* The documentation has been updated regarding the values that the mixture of a piston engine can be set to (PR #1048 and issue #1037).
* Fixed a bug where a telnet connection could not be reopened (PR #1078)
* Added the ability to change the location of thrusters via properties (PR #1073)
* The error management in socket code has been improved to intercept and report more errors. This should help the user finding errors more easily when using this feature (PR #1081)
* Added the ability to modify the property `ic/vw-mag-fps` i.e. the initial value of the wind velocity magnitude (PR #1093)
* Reduced the console output when debug level is zero (PR #1092)
* The thrust augmentation command is now clamped to the range [0.0 - 1.0] (PR #1090)
* Added properties `AGL-ft` to display the height above the ground of each contact (PR #1104)
* Clarified the document of `<function>` to state that it can contain a single `<table>` element (PR #1131)

#### Distributed flight dynamics models
* The model of the [Short S23 Empire](https://en.wikipedia.org/wiki/Short_Empire) has been updated.
* Fixed the initial conditions of the script [ball_orbit.xml](https://github.com/JSBSim-Team/jsbsim/blob/e5192df5b7e6f4003fdc351bde17d38dfc6b1bff/scripts/ball_orbit.xml) so that its orbit gets circular (i.e. its eccentricity is zero). See discussion #985 for more details.
* The model of the [Zeppelin NT airship](https://en.wikipedia.org/wiki/Zeppelin_NT) a.k.a ZLT-NT has been updated.
* The model of the [Submarine Scout airship](https://en.wikipedia.org/wiki/SS_class_airship) has been updated.
* The license of the following files have been changed from GPLv2+ to LGPLv2.1+
  * [systems/airship-mooring.xml](https://github.com/JSBSim-Team/jsbsim/blob/e5192df5b7e6f4003fdc351bde17d38dfc6b1bff/systems/airship-mooring.xml)
  * [systems/airship_added_mass.xml](https://github.com/JSBSim-Team/jsbsim/blob/e5192df5b7e6f4003fdc351bde17d38dfc6b1bff/systems/airship_added_mass.xml)
  * [systems/hydrodynamic-planing-floats.xml](https://github.com/JSBSim-Team/jsbsim/blob/e5192df5b7e6f4003fdc351bde17d38dfc6b1bff/systems/hydrodynamic-planing-floats.xml)
  * [systems/hydrodynamic-planing.xml](https://github.com/JSBSim-Team/jsbsim/blob/e5192df5b7e6f4003fdc351bde17d38dfc6b1bff/systems/hydrodynamic-planing.xml)
  * [systems/hydrodynamics.xml](https://github.com/JSBSim-Team/jsbsim/blob/e5192df5b7e6f4003fdc351bde17d38dfc6b1bff/systems/hydrodynamics.xml)
  * [systems/visual-reference-point-extensions.xml](https://github.com/JSBSim-Team/jsbsim/blob/e5192df5b7e6f4003fdc351bde17d38dfc6b1bff/systems/visual-reference-point-extensions.xml)

#### Python module
* Improved the caching of property nodes for faster lookups (PR #993)
* The reference counting of property nodes is now shared between the C++ code and the python module. This avoids crashes due to the inadvertent deletion of pointers by the C++ code of pointers that are also used by the Python module. (PR #1000)
* Fixed the documentation of the methods `FGPropertyNode.get_double_value` and `FGPropertyNode.set_double_value` (PR #1048 and issue #922).
* Made the Python module compatible with NumPy 2.0 (PR #1107)

#### Aeromatic++
* Fixed a typo in a message printed in the console by aeromatic++ (PR #1002)
* Fixed the headers so that aeromatic++ can also be compiled with VS2015 (PR #1020)

### Developer
* The CMake target `libJSBSim` now includes the list of the headers it uses for cross reference in other CMake files (PR #1022).
* Some maintenance / minor changes have been done (see PR #1031 and PR #1050 for details).
* Replaced hardcoded buffer lengths by a computation of the length using `sizeof()`. This avoids potential mismatches in future updates of the files (PR #1035).
* JSBSim is now compiled and tested on MacOS M1 runners (PR #1030)
* `libexpat` has been updgraded to 2.6.2 (PR #1080)
* The XML element of late bound properties is now released after successful binding for memory saving (PR #1096)

**Comment by bcoconni** [comment]
Note that:
* The general use of PRs make the changelog easier to write :smile:
* I have purposely skipped the PR #1094 (logging system revamp) from the release 1.2.1 as I consider it's still work in progress.

**Reply by seanmcleod** [reply]
> ...as I consider it's still work in progress.

I'm sure delayed because of all the time you're spending at the Olympics? 😉

**Reply by bcoconni** [reply]
> I'm sure delayed because of all the time you're spending at the Olympics? 😉

Not *at* the Olympics but watching them definitely :wink:

**Comment by seanmcleod** [comment]
Looks like a good and comprehensive summary.

**Comment by bcoconni** [comment]
Thanks, so I went ahead and v1.2.1 is now released.
